### PR TITLE
feat(api): Add pagination metadata to Event module

### LINKED ISSUE #336: API: Add pagination metadata
## Description

Add the following fields when returning paginated response in every module:
- `page`: The current page number
- `perPage`: The number of items per page
- `pageCount`: The total page count
- `totalCount`: The total number of items present
- `links` (All links to have the relative path. For example: `/api/environment/all/:projectId?<query>`)
  - `self`: The link to this current page
  - `first`: The link to the first page
  - `previous`: The link to the previous page
  - `next`: The link to the next page
  - `last` : The link to the last page

We want the computation to be done in the service classes.

## Solution

- Create a utility function under `common` module and name it `paginate`. This function should take in the page details and generate the pagination metadata
- Update all the functions in the service layer that return paginated data to return the data in `items` and the metadata in `metadata`
- Update the E2E test of the module to reflect on the changes

## Objectives

This issue is the parent issue. Below you will find a list of issues that relate to this. These issues encompass the scope of change in every module:

- #345
- #337
- #338
- #339
- #340
- #341
- #342
- #343
- #344

### LINKED ISSUE #344: API: Add pagination metadata to Event module
## Description

Refer to #336 

## Scope of change

- https://github.com/keyshade-xyz/keyshade/blob/bc15eadf33a4ddc8ed496decd16e979aaf71ada9/apps/api/src/event/service/event.service.ts#L13-L58

### PR DESCRIPTION
**Pull request by muntaxir4**
## Description

- Add support for paginated response for `getEvents()` of event.service.ts under API.
- Add and improve relevant test cases.
- Modify event variable's value in affected modules' e2e, where state of events are queried.

Fixes #344

Also, Fixes #336 after merging this and #382 #387 #388 #389 #390 #391 #393.

## Dependencies
N/A

## Future Improvements
N/A

## Screenshots of relevant screens
Extra test:
image

## Developer's checklist

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-check on my work

**If changes are made in the code:**

- [x] I have followed the [coding guidelines](https://google.github.io/styleguide/jsguide.html)
- [x] My changes in code generate no new warnings
- [ ] My changes are breaking another fix/feature of the project
- [x] I have added test cases to show that my feature works
- [x] I have added relevant screenshots in my PR
- [x] There are no UI/UX issues

### Documentation Update

- [ ] This PR requires an update to the documentation at [docs.keyshade.xyz](https://docs.keyshade.xyz)
- [x] I have made the necessary updates to the documentation, or no documentation changes are required.

### CODE REVIEW

**Comment by muntaxir4** [comment]
/describe The bot is asleep ig :disappointed:

File: apps/api/src/project/project.e2e.spec.ts
Comment (rajdip-b) [inline apps/api/src/project/project.e2e.spec.ts:1]: Okay this PR has many unrelated changes in the tests.

**Review by rajdip-b — CHANGES_REQUESTED** [review]

File: apps/api/src/project/project.e2e.spec.ts
Comment (muntaxir4) [reply apps/api/src/project/project.e2e.spec.ts:1]: Can you mention which one? The e2e tests will give errors if I do not update other modules.
I just changed `response.json()[0]` -> `rsponse.json().items[0]` while querying events.

Comment (rajdip-b) [reply apps/api/src/project/project.e2e.spec.ts:1]: Yes, I understand that. My point is, since you have put up other PRs aswell, it will be better to make changes to the test in the individual PRs rather than this one pr

Comment (muntaxir4) [reply apps/api/src/project/project.e2e.spec.ts:1]: I previously thought it could cause merge conflicts. Just checked in forked repo those are independent. Also, removed those as dependencies.

Comment (rajdip-b) [reply apps/api/src/project/project.e2e.spec.ts:1]: Nah it wont be causing merge conflcts. We just dont want one commit to be dependent on another commit. Thats why, Im asking you to couple up the test changes along with the respective PRs

Comment (muntaxir4) [reply apps/api/src/project/project.e2e.spec.ts:1]: To include `response.json().items[0]` in each module's test, the event module needs to have paginated response. Hence, doing this directly will throw error while committing.

Comment (rajdip-b) [reply apps/api/src/project/project.e2e.spec.ts:1]: Stupid me! my brain stopped working haha! This works fine then :D

**Review by rajdip-b — APPROVED** [review]
LGTM!
